### PR TITLE
swiping left or right to navigate between songs

### PR DESCRIPTION
**Pull request by MiTo0o**

### CODE REVIEW

File: src/test/App.test.tsx [hunk 1]
```diff
@@ -273,6 +274,49 @@ describe("App", () => {
...
+  //   await page.mouse.up();
+  //   // doesn't move
+  //   expect(page.url()).toEqual(getSongLink(6));
+
+  //   await page.mouse.move(100, 200);
+  //   await page.mouse.down();
+  //   await page.mouse.move(200, 200);
+  //   await page.mouse.up();
+
+  //   expect(page.url()).toEqual(getSongLink(6));
```
Comment (MiTo0o) [inline src/test/App.test.tsx:315]: welp, I tried to write a test but it just won't work. I think I spent more time trying to the test to work than I did working on the feature. the mouse up and mouse move is working, I can see the position changing by logging it out. But the simulated swiping just isn't working. No matter how I simulate the swipe, it always navigates to the previous song/hymn.

Comment (MiTo0o) [reply src/test/App.test.tsx:315]: just tried to upgrade puppeteer and use the new mouse drag function, it's exceeding time limit of the test even when I put 50 sec on the test. I'm kinda out of ideas, maybe we could just simulate mouse dragging with pure js for the test.

image

Comment (MiTo0o) [reply src/test/App.test.tsx:315]: thanks dad, I tested out `drag()` but it was timing out. I'll look into the other drag function

**Comment by MiTo0o** [comment]
closes #106

**Review by brotatotes — CHANGES_REQUESTED** [review]
Thanks for trying to get the UT working. I think in this case we can forgo the UT. But let's do two things:
- let's enable this feature on mobile only. I don't think there's any reason to have it on Desktop, and could be confusing if people triggered it accidentally on Desktop. 
- let's test it manually on a mobile device to ensure the thresholds and constants are configured in a way that triggers on natural swiping and isn't too sensitive or too insensitive.

File: src/pages/SongPage.tsx [hunk 2]
```diff
@@ -35,6 +48,33 @@ const SongPage: React.FC = () => {
...
+    if (isDesktop()) {
+      return;
+    }
+    const gesture: Gesture = createGesture({
+      el: document.getElementById("song-page-body") as Node,
+      threshold: SWIPE_THRESHOLD,
+      gestureName: "swipe-gesture",
+      direction: "x",
+      onEnd: (detail: GestureDetail) => {
+        console.log(detail);
```
Comment (brotatotes) [inline src/pages/SongPage.tsx:61]: remove?

File: src/pages/SongPage.tsx [hunk 3]
```diff
@@ -35,6 +48,33 @@ const SongPage: React.FC = () => {
     };
   }, [bookId]);
 
+  useEffect(() => {
+    if (isDesktop()) {
+      return;
+    }
+    const gesture: Gesture = createGesture({
```
Comment (brotatotes) [inline src/pages/SongPage.tsx:55]: can we abstract out this logic to some Utils so we also don't have to import the constants here?

File: src/test/App.test.tsx (hunk 1 above)
Comment (brotatotes) [reply src/test/App.test.tsx:335]: can remove i think

**Review by brotatotes — CHANGES_REQUESTED** [review]

**Review by brotatotes — APPROVED** [review]

**Pull request merged**